src/Storages/ObjectStorage/DataLakes/Iceberg/Mutations.cpp

```diff
@@ -262,7 +263,7 @@ std::optional<WriteDataFilesResult> writeDataFiles(
                 if (!update_data_statistics.contains(partition_key))
                     update_data_statistics.emplace(partition_key, DataFileStatistics(data_schema->getArray(Iceberg::f_fields)));
 
-                if (!update_data_writers.contains(partition_key))
+                if (auto it = update_data_writers.find(partition_key); it != update_data_writers.end())
```

Comment:
This looks incorrect, I think need to change to 
```
auto it = update_data_writers.find(partition_key); 
if (it == update_data_writers.end())
{
    ...
    it = update_data_writers.emplace(partition_key, std::move(data_output_format));
}
...
it.second->write(data_block);

```
(currently it looks it like will segfault)